File: longest-common-prefix/longest-common-prefix.cpp
class Solution {
  
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        if (strs.size() == 0)
            return "";
        string prefix = strs[0];
        
        for(int i = 1; i < strs.size(); ++i) {
            if(strs[i].find(prefix) != 0) {
                while(strs[i].find(prefix) != 0) {
                    prefix = prefix.substr(0, prefix.size()-1);
                    if(prefix.size() == 0) {
                        return "";
                    }
                }
            }    
        }
        
        return prefix;
    }
    
    
};

File: longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        str_len = [len(s) for s in strs]
        min_l = min(str_len)
        prefix = ""
        for i in range(min_l):
            c = strs[0][i]
            for j in range(1, len(strs)):
                if strs[j][i] != c:
                    return prefix
            prefix = prefix + c
        return prefix